app/scripts/lib/errors.js

```diff
@@ -100,15 +100,25 @@ define([
      * @param {String} [context]
      */
     toError: function (type, context) {
-      var message = this.toMessage(type);
-      if (! message) {
-        message = this.toMessage('UNEXPECTED_ERROR');
-      }
+      var errno = this.toErrno(type);
+      var message = this.toMessage(errno);
 
       var err = new Error(message);
 
+      if (typeof type === 'object') {
```

Comment:
_.extend will pull in items from `type`s prototype as well, I didn't want that in case `type` was already an `Error` and we start copying over functions on the error's prototype. Maybe what I should do is 

```js
if (typeof type === 'object' && ! (type instanceof Error)) { 
  _.extend(...
```